Spokestack/NLUTensorflowMetadata.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  NLUTensorflowMetadata.swift
+//  Spokestack
+//
+//  Created by Noel Weichbrodt on 1/29/20.
+//  Copyright © 2020 Pylon AI, Inc. All rights reserved.
+//
+
+import Foundation
+
+/// Provides model-specific structured access to the NLU model metadata.
+internal struct NLUTensorflowMeta {
+    private let configuration: SpeechConfiguration
+    internal let model: NLUTensorflowMetadata
+    
+    /// Using the provided configuration that includes the `nluModelMetadataPath`, hydrate the corresponding the metadata structure.
+    /// - Parameter configuration: The global SpeechConfiguration.
+    init(_ configuration: SpeechConfiguration) throws {
+        self.configuration = configuration
+
+        let metadataURL = URL(fileURLWithPath: configuration.nluModelMetadataPath)
+        let metaData = try Data(contentsOf: metadataURL)
+        guard let metadata = try? JSONDecoder().decode(NLUTensorflowMetadata.self, from: metaData) else {
+            throw NLUError.metadata("Could not parse model metadata file set at nluModelMetadataPath.")
+        }
+        self.model = metadata
+    }
+}
+
+/// The internal nlu model metadata structure.
+internal struct NLUTensorflowMetadata: Codable {
+    let intents: [NLUTensorflowIntent]
+    let tags: Array<String>
+}
+
+internal struct NLUTensorflowIntent: Codable {
+    let name: String
+    let slots: [NLUTensorflowSlot]
+}
+
+internal struct NLUTensorflowSlot: Codable {
+    let name: String
+    let type: String
+    let facets: String
+}
+
+internal struct NLUTensorflowSelset: Codable {
```

Comment:
Any change to the fascet type/representation/parsing is going to be a point release due to the overall coupling of the system no matter the extensibility of the slot fascet decoders.

Reply:
Sure; I was thinking more of the difficulty of adding new slot types (can an end user even do this?), not so much a change to the existing schemas.

Reply:
In the android version, the slot parser interface would allow for a user-provided (eg one not included in the Spokestack package) slot parser to be configured?

Reply:
Yep; you just have to call `registerSlotParser` with a slot type and the name of your parser class (which has to implement `SlotParser`) when you're building the NLU object.